woocommerce/woocommerce
# [CYS] Update the call to the ai product endpoint to account for the newly introduced last_product param.

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:



Update the call to the ai product endpoint to account for the newly introduced last_product param. This PR complements the changes made on https://github.com/woocommerce/woocommerce-blocks/pull/11971 , ensuring the rewrite rules are flushed only when the last product is updated and not one per product. 

Additionally, the `index` param was removed for the '/wc/private/ai/product' endpoint since its not required anymore. This private API is used on a feature that hasn't been released yet (the CYS flow), so we can safely remove the param without worrying about deprecating.



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Create a new WooCommerce installation with this version.
2. Install and activate the WooCommerce Blocks plugin from this zip file: [woocommerce-gutenberg-products-block.zip](https://github.com/woocommerce/woocommerce/files/13527255/woocommerce-gutenberg-products-block.zip)
3. Ensure you have the WooCommerce Beta Tester plugin installed.
4. Head over to `/wp-admin/tools.php?page=woocommerce-admin-test-helper` and enable `customize-store` feature flag:

<img width="1233" alt="Screenshot 2023-10-22 at 10 32 24" src="https://github.com/woocommerce/woocommerce-blocks/assets/15730971/816867c6-34e5-40a1-a87d-f4a7cda46429">

5. Go to WooCommerce -> Home and click Customize Store task
6. Click Design with AI button
7. Fill out a business description and proceed
8. Make sure everything works as expected, and you can have your store with AI-generated content without problems.



### Changelog entry




<details>

#### Significance




#### Type




#### Message 

Customize Your Store: Update the call to the the AI product endpoint to add the new last_product param to the request.

#### Comment 

</details>
